jaxrs-service/src/main/java/com/quorum/tessera/admin/ConfigResource.java

```diff
@@ -85,4 +90,49 @@ public Response getPeers() {
         }).build();
     }
 
+    @GET
+    @Path("/keypairs/{publicKey}")
+    public Response getKeyPair(@PathParam("publicKey") String base64PublicKey) {
+
+        Base64.Decoder base64Decoder = Base64.getDecoder();
+
+        PublicKey publicKey = PublicKey.from(base64Decoder.decode(base64PublicKey));
+
+        Set<PublicKey> publicKeys = configService.getPublicKeys();
+
+        if(!publicKeys.contains(publicKey)) {
+            throw new NotFoundException("No key pair found with public key " + base64PublicKey);
+        }
+
+        //TODO use dedicated transport object instead?
+        KeyData responseData = new KeyData();
+        responseData.setPublicKey(base64PublicKey);
+        responseData.setPrivateKey("REDACTED");
+
+        return Response.ok(responseData).build();
+    }
+
+    @GET
+    @Path("/keypairs")
+    public Response getKeyPairs() {
+        Set<PublicKey> publicKeys = configService.getPublicKeys();
+
+        if(publicKeys.isEmpty()) {
+            throw new NotFoundException("No key pairs found");
+        }
+
+        //TODO use dedicated transport object instead?
+        List<KeyData> keyPairData = publicKeys.stream()
+            .map(PublicKey::encodeToBase64)
+            .map(publicKey -> {
+                KeyData kd = new KeyData();
+                kd.setPublicKey(publicKey);
+                kd.setPrivateKey("REDACTED");
```

Comment:
How about creating a new object that only contains the public key? The "REDACTED" and type in the response isn't very useful for the user.

Reply:
This was something I had considered doing - I'll add it in

Reply:
Noticed you mentioned the same for the single key endpoint. I think it is worth doing; if you don't fancy using a new object you can create the JSON response directly.